### PR TITLE
Issue 401: Adding Info Icon to Priority Title in Map Legend

### PR DESCRIPTION
**Pull request by mskNeo**
https://github.com/CodeForPhilly/vacant-lots-proj/issues/401

Added the Phosphor Info Icon to the map legend. Had to hardcode it in CSS styling to get the specific icon from Phosphor Icons.

### CODE REVIEW

**Comment by paulhchoi** [comment]
@mskNeo your PR covers the scope of this ticket, but for a more general UI/UX question: @thansidwell should hovering/clicking this icon show a tooltip? Just having an un-interactable 'i' icon feels like it may lead to some confusion, so if a tooltip should be added, I suggest we hold this PR as a draft until the tooltip is ready to be added.

**Comment by thansidwell** [comment]
@paulhchoi Yes, we had intended when a user hovers over the icon a tooltip would show up explaining how we determine priority. We still don't have that written yet.

**Comment by mskNeo** [comment]
I can try implementing a tooltip, but do we have any specific UI mockups/vision for it? @thansidwell

**Comment by thansidwell** [comment]
@mskNeo That would be great if you could. There are a few other places we are using it. We are using the NextUI component library. Here's their tooltip. It's very close to what I designed.  https://nextui.org/docs/components/tooltip

Here's the documentation of what it should look like. https://www.figma.com/file/NAFkgq34abW6uJ0R7PW24T/Prototype---Clean-%26-Green-Philly?type=design&node-id=1435%3A8475&mode=design&t=rwvoZG2F2y3A2hfM-1

@brandonfcohen1 @paulhchoi Do we have a tooltip already?

**Comment by brandonfcohen1** [comment]
> @mskNeo That would be great if you could. There are a few other places we are using it. We are using the NextUI component library. Here's their tooltip. It's very close to what I designed. https://nextui.org/docs/components/tooltip
> 
> Here's the documentation of what it should look like. https://www.figma.com/file/NAFkgq34abW6uJ0R7PW24T/Prototype---Clean-%26-Green-Philly?type=design&node-id=1435%3A8475&mode=design&t=rwvoZG2F2y3A2hfM-1
> 
> @brandonfcohen1 @paulhchoi Do we have a tooltip already?

yeah we've been using nextui tooltip

**Comment by bacitracin** [comment]
Hey @mskNeo ! When you add the NextUI tooltip, could you please add 2 quick a11y things?

1. Make the info icon keyboard accessible. So you should be able to TAB to the info icon, focus on it, and the tooltip will appear. Most likely all you need is to add a `tabIndex="0"` to the icon.
2. Add an alt text to the icon.

Here is a previous PR where I added these things https://github.com/CodeForPhilly/vacant-lots-proj/pull/350

Let me know if you have any questions!

**Comment by mskNeo** [comment]
@bacitracin it seems like the map legend is autogenerated so adding the NextJS tooltip won't work since the legend elements are not in the codebase but rather rendered by the library. I'll try making a custom legend. @thansidwell are there any features of the legend as it is now that are being used? like collapsing it or anything?

**Comment by brandonfcohen1** [comment]
> @bacitracin it seems like the map legend is autogenerated so adding the NextJS tooltip won't work since the legend elements are not in the codebase but rather rendered by the library. That's why I used a CSS approach to add the info icon. I'll try making a custom legend. @thansidwell are there any features of the legend as it is now that are being used? like collapsing it or anything?

+1 on writing a custom legend

File: package.json [hunk 1]
```diff
@@ -19,7 +19,6 @@
     "framer-motion": "^10.16.4",
     "lodash": "^4.17.21",
     "mapbox-gl": "^3.2.0",
-    "mapboxgl-legend": "^1.14.0",
```
Comment (mskNeo) [inline package.json:22]: removing the legend library since we have a custom component now

File: src/app/mapLegend.css [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+  opacity: 1;
+}
+.mapboxgl-ctrl-bottom-right .mapboxgl-ctrl-legend,
+.mapboxgl-ctrl-top-right .mapboxgl-ctrl-legend {
+  align-items: flex-end;
+}
+.mapboxgl-ctrl-bottom-right .mapboxgl-ctrl-legend,
+.mapboxgl-ctrl-bottom-left .mapboxgl-ctrl-legend {
+  flex-direction: column-reverse;
+}
```
Comment (mskNeo) [inline src/app/mapLegend.css:232]: these are all from the mapboxgl-legend library, just copied all the CSS styles

File: src/components/MapLegend.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    this._container.innerHTML = renderToStaticMarkup(
+      <MapLegend {...options} />
+    );
+  }
+
+  // able to add event listeners here for interactivity with layer
+  onAdd = (map: MapboxMap) => {
+    this._map = map;
+    return this._container;
+  };
```
Comment (mskNeo) [inline src/components/MapLegend.tsx:32]: now we should be able to add event listeners for the map, expand functionality and maybe add interactivity

File: src/app/mapLegend.css
Comment (mskNeo) [inline src/app/mapLegend.css:1]: Got all these styles from the mapboxgl-legend we used to use

File: src/components/MapLegendControl/MapLegend.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    this._container.innerHTML = renderToStaticMarkup(
+      <MapLegend {...layerStyle} />
+    );
+  }
+
+  // able to add event listeners here for interactivity with layer
+  onAdd = (map: MapboxMap) => {
+    this._map = map;
+    return this._container;
+  };
```
Comment (mskNeo) [inline src/components/MapLegendControl/MapLegend.tsx:113]: now able to add event listeners to map for legend, enabling interactivity

**Review by brandonfcohen1 — CHANGES_REQUESTED** [review]
@mskNeo deployment is failing- looks like you need to merge in the latest `staging`

File: src/components/PropertyMap.tsx [hunk 5]
```diff
@@ -235,14 +243,19 @@ const PropertyMap: FC<PropertyMapProps> = ({
...
+      const legendSummary = document.getElementById("legend-summary");
+      if (legendSummary) {
+        summaryInfo = createPortal(
+          <Tooltip content="We prioritize properties based on how much they can reduce gun violence considering the vacancy, gun violence, cleanliness, and tree canopy.">
+            <Info
+              alt="Priority Info"
+              className="text-gray-500 cursor-pointer"
+              tabIndex={0}
+            />
+          </Tooltip>,
```
Comment (mskNeo) [inline src/components/PropertyMap.tsx:256]: Screenshot 2024-03-24 at 6 11 11 PM
This is what it looks like now, should I make the tooltip a smaller width?

**Comment by mskNeo** [comment]
> @mskNeo deployment is failing- looks like you need to merge in the latest `staging`

@brandonfcohen1  I synced my forked repo with staging and merged but it's still failing it seems

**Comment by thansidwell** [comment]
@mskNeo I noticed we have two different style tooltips. On the live site and your tooltip. Are you using a different tooltip than what's used on the 'Download' and 'View' buttons? And both don't match the prototype. 

CleanShot 2024-03-25 at 11 59 18@2x

We should just be using one toolip component and it should look like the prototype. https://www.figma.com/file/NAFkgq34abW6uJ0R7PW24T/Prototype---Clean-%26-Green-Philly?type=design&node-id=1435%3A8475&mode=design&t=in6whOv1xRCIDRgL-1 

Can you fix that on this ticket or should I just write a new one?

**Comment by mskNeo** [comment]
> > > @mskNeo deployment is failing- looks like you need to merge in the latest `staging`
> > 
> > 
> > @brandonfcohen1 I synced my forked repo with staging and merged but it's still failing it seems
> 
> here is the error- are you sure you have the latest `staging`? you can run `npm run build` locally to test the build before deploying
> 
> ```
> 
> Failed to compile.
> ./src/components/PropertyMap.tsx:41:23
> Type error: Cannot find module './ZoomModal' or its corresponding type declarations.
>   39 | import MapboxGeocoder from "@mapbox/mapbox-gl-geocoder";
>   40 | import "@mapbox/mapbox-gl-geocoder/dist/mapbox-gl-geocoder.css";
> > 41 | import ZoomModal from "./ZoomModal";
>      |                       ^
>   42 | import { Coordinates } from "../app/types";
>   43 | import { MapLegendControl } from "./MapLegendControl";
>   44 | import { createPortal } from "react-dom";
> Error: Command "npm run build" exited with 1
> ```

@brandonfcohen1 My forked repository for some reason wasn't syncing properly with the staging branch, thankfully my main branch did so I had to manually check out the missing files. Hopefully that works

**Comment by brandonfcohen1** [comment]
legend looks great. Will let @thansidwell approve the UI

Is it intended that clicking the `i` icon minimizes the legend?

**Comment by mskNeo** [comment]
> legend looks great. Will let @thansidwell approve the UI
> 
> Is it intended that clicking the `i` icon minimizes the legend?

@brandonfcohen1 clicking the upper bar of the legend in general minimizes, not just the I. So clicking on the empty space to the right of it closes it too. This is the same behavior as the library legend that we had before.

**Comment by brandonfcohen1** [comment]
> > legend looks great. Will let @thansidwell approve the UI
> > Is it intended that clicking the `i` icon minimizes the legend?
> 
> @brandonfcohen1 clicking the upper bar of the legend in general minimizes, not just the I. So clicking on the empty space to the right of it closes it too. This is the same behavior as the library legend that we had before.

haha nice, had never clicked that before

**Review by brandonfcohen1 — APPROVED** [review]
approved by me, awaiting @thansidwell

**Comment by thansidwell** [comment]
It looks good to me! 

I'll write another ticket to make further styling updates to the tooltip.

**Pull request merged**